libraries/lib-math/LinearFit.h

```diff
@@ -0,0 +1,41 @@
+#include <cassert>
+#include <numeric>
+#include <utility>
+#include <vector>
+
+template <typename T, typename U>
+std::pair<double, double> LinearFit(
```

Comment:
You use template parameters T and U, not requiring them to be the same.

You use `double` in the return type.

Maybe you might make this more general?  (Could there be some future need to use it with `long double` or `std::complex<float>?)

Maybe

`template<typename ResultType = std::common_type_t<double, T, U>, typename T, typename U>`

Maybe also the common cases where T and U are both double or both float could also declare `extern MATH_API` explicit template instantiations here, and a companion .cpp file could generate the definitions.

Reply:
I do think the case where `x` is of integer type is quite common. That's the use case I had, rounding onsets to tatum (or "tick") indices.
For the rest, it's a bit complicated to find a solution that satisfies all cases without compiler warnings. I just generalized the weight type for usage convenience, but the return type is just double. Users who need float coefficients shouldn't have trouble downcasting.